src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -479,25 +479,54 @@ private void SetOpcode(DynamicTraceEventData template, int opcode)
         private DynamicTraceEventData CreateTemplate(EventPipeEventMetaDataHeader eventMetaDataHeader)
         {
             string opcodeName = ((TraceEventOpcode)eventMetaDataHeader.Opcode).ToString();
-
             int opcode = eventMetaDataHeader.Opcode;
             if (opcode == 0)
```

Comment:
Seeing code snippets that set eventMetadataHeader.EventName and compute the substrings multiple times suggest this could probably be refactored to be less duplicative. For example:
````C#
            if (opcode == 0)
            {
                GetOpcodeFromEventName(eventMetaDataHeader.EventName, out opcode, out opcodeName);
            }
            FilterOpcodeNameFromEventName(eventMetaDataHeader, opcode);
````